libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/ManeuverProcessor.kt

```diff
@@ -390,19 +405,28 @@ internal object ManeuverProcessor {
                     list.forEach {
                         val directions = it.directions()
                         val active = it.active()
+                        val activeDirection: String? = profile?.let { value ->
+                            // In case of PROFILE_CYCLING [BannerComponent#activeDirection] is
+                            // unavailable and hence we need to fallback to [BannerText#modifier]
+                            if (active!! && value == DirectionsCriteria.PROFILE_CYCLING) {
```

Comment:
The assumption that `active` is present is only true for `Lane` component types which we're not checking above as far as I can tell.

Reply:
I didn't understand what you meant here?

Reply:
You're force-casting `active` from a nullable type to a non-nullable type here.

`active` can only be non-null if the component type is `Lane`, which you're not checking above. If some other component type ended up in this code path, we could crash. And again, with defensive programming in mind, why not check the nullability before using the type?

Reply:
I am already checking if the type is `lane` using the if condition below
```
private fun getLaneGuidance(
        profile: String?,
        drivingSide: String,
        bannerInstruction: BannerInstructions
    ): Lane? {
        val subBannerText = bannerInstruction.sub()
        return subBannerText?.let { subBanner ->
            if (subBanner.type() == null && subBanner.text().isEmpty()) {
```
This is how the response was earlier. There was no `lane` type. Rather if the type was `null` we could say that it is a lane. I raised this as a bug a few months back and looks like directions fixed it. So now I need to add `StepManeuverType.Lane` to MapboxJava and then use it here. 

As far as `active` is concerned; this can never be null. I am not sure why is it annotated as `Nullable` in `Mapbox-Java`. Same goes for `LegStep.drivingSide`. Every field in MapboxJava is annoatated as nullable but if you look at directions documentation there are fields that can never be null. Also, if I check if (active == null), there is no defined behavior for it. This is a problem in general in our codebase I think, that we are making a lot of assumptions based on nullability from directions response and deciding the behavior, where is some cases we should just throw exceptions. 

So, I don't think there is any need for checking the nullability of `active` and I think I would just go forward with it.

Reply:
> As far as active is concerned; this can never be null.

It can be null if the component type is not `Lane`. This is my only concern.

As far as I see there's a `BannerComponent#type` which can be `Lane` (as visible in your capture), and only for this type the `BannerComponent#active` field is available, which we're not checking before coercing `active` to a non-nullable type.

Basically instead of checking `BannerSection#type == null`, we should be checking `BannerSection#components#type == Lane`, shouldn't we?

If there was in the future introduce a `BannerSection` which passed
```
subBanner.type() == null && subBanner.text().isEmpty()
``` 
but actually wasn't a `Lane`, we would start crashing.

Reply:
Added a check on `type == lane`.

/cc @LukasPaczos